r/R/dataset-format.R

```diff
@@ -228,9 +225,32 @@ check_csv_file_format_args <- function(...) {
       call. = FALSE
     )
   }
+}
+
+get_opt_names <- function(args) {
+  opt_names <- names(args)
+
+  # extract names of parse_options, read_options, and convert_options
+  if ("parse_options" %in% names(args) && is.list(args[["parse_options"]])) {
```

Comment:
`args[["parse_options"]]` can either be a list of options, or a `CsvParseOptions` object. This helper function is only designed to validate the former; the latter is more advanced/off-label usage and it's up to users to safely pass the correct arguments through here (you could argue that this shouldn't be the case, but doing it this way is consistent with how we've done this in other bits of the codebase).

I thought about documenting this, but I don't know *where* it would go - I don't think inside this function?

Reply:
`parse_options = CsvParseOptions$create(...)` is the way that users have been doing things thus far as passing in lists is new - this additional way of doing things only got merged in in the past few weeks.  I'm not sure what the right call is here; I don't want to error on passing in a CsvParseOptions object and break folks' existing code but do want to encourage passing in options as a list so it actually hits our validation code.

Reply:
I added more info here, and more explicit acknowledgement of this to `CsvFileFormat$create()`